src/test_c2s_srv.c

```diff
@@ -548,12 +619,18 @@ int test_c2s(Connection *ctl, tcp_stat_agent *agent, TestOptions *testOptions,
     // Set up the FD_SET and activeStreams for the next select.
     activeStreams = connections_to_fd_set(c2s_conns, streamsNum, &rfd, &max_fd);
   }
-
   measured_test_duration = secs() - start_time;
+  // From the NDT spec:
   //  throughput in kilo bits per sec =
   //  (transmitted_byte_count * 8) / (time_duration)*(1000)
   *c2sspd = (8.0e-3 * bytes_read) / measured_test_duration;
 
+  // As a kindness to old clients, drain their queues for a second or two.
+  // TODO: Fix client code to eliminate the need for this.  Clients that need
+  //       this line should be removed from their respective gene pools and
+  //       this code should be deleted.
```

Comment:
Unfortunately, most clients will likely trigger this line. The question is not "will an old client trigger this line", but instead "if we just closed the socket would the client crash". When writing to a closed socket, `web100clt` will straight-up crash. The node.js client will call its `onclose()` method that has actual error handling, so it won't break. The only way to know whether the client needs these extra `read()`s is to close early and see if the client crashes :/ 

Since that's not the kind of "testing" we want, I think we are stuck. We could log the client identifier, and then keep track of how many of them look like `web100clt`, but that doesn't seem like a job for this PR.